applications/GeoMechanicsApplication/tests/cpp_tests/test_dgeosettlement.cpp

```diff
@@ -0,0 +1,118 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#include "testing/testing.h"
+#include "custom_workflows/dgeosettlement.h"
+#include "input_utility_stub.h"
+
+using namespace Kratos;
+
+namespace Kratos::Testing {
+
+const std::string parameter_json_settings = "{"
+                                            "\"solver_settings\":"
+                                            "{"
+                                            "\"model_part_name\":\"test\","
+                                            "\"model_import_settings\":"
+                                            "{"
+                                            "\"input_type\": \"mdpa\","
+                                            "\"input_filename\": \"mesh_stage1\""
+                                            "},"
+                                            "\"material_import_settings\": "
+                                            "{"
+                                            "\"materials_filename\": \"MaterialParameters1.json\""
+                                            "}"
+                                            "}"
+                                            "}"; // these have material_import settings
+
+
+void RunStage(KratosGeoSettlement& rSettlement) {
+    rSettlement.RunStage("",
+                         "",
+                         [](const char *) {}, // kept empty as a stub method
+                        [](const double) {}, // kept empty as a stub method
+                        [](const char *) {}, // kept empty as a stub method
+                        []() { return true; });
```

Comment:
I think `return false;` is more appropriate here (you don't want to cancel the operation).